b9lab-content/4-running-a-chain/04-node-api-and-cli.md

```diff
@@ -1 +1,464 @@
+
+```go
+func NewRootCmd() (*cobra.Command, params.EncodingConfig)
+```
+
+In it, [basic properties](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/simd/cmd/root.go#L51-L53) such as the application name are defined:
+
+```go
+rootCmd := &cobra.Command{
+    Use:   "simd",
+    Short: "simulation app",
+```
+
+In addition, observe that cobra is imported and [used for the CLI](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/simd/cmd/root.go#L144-L155) to redirect:
+
+```go
+rootCmd.AddCommand(
+    genutilcli.InitCmd(simapp.ModuleBasics, simapp.DefaultNodeHome),
+    genutilcli.CollectGenTxsCmd(banktypes.GenesisBalancesIterator{}, simapp.DefaultNodeHome),
+    genutilcli.MigrateGenesisCmd(),
+    genutilcli.GenTxCmd(simapp.ModuleBasics, encodingConfig.TxConfig, banktypes.GenesisBalancesIterator{}, simapp.DefaultNodeHome),
+    genutilcli.ValidateGenesisCmd(simapp.ModuleBasics),
+    AddGenesisAccountCmd(simapp.DefaultNodeHome),
+    tmcli.NewCompletionCmd(rootCmd, true),
+    NewTestnetCmd(simapp.ModuleBasics, banktypes.GenesisBalancesIterator{}),
+    debug.Cmd(),
+    config.Cmd(),
+)
+```
+
+Also explain [`simapp/app.go`](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/app.go) in which each module and keykeeper will be imported. The first thing you will see is a considerable [list of modules](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/app.go#L33-L44) that are used by most cosmos-sdk apps:
+
+```go
+...
+  "github.com/cosmos/cosmos-sdk/x/auth"
+  "github.com/cosmos/cosmos-sdk/x/auth/ante"
+  authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+  authsims "github.com/cosmos/cosmos-sdk/x/auth/simulation"
+  authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+  authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+  "github.com/cosmos/cosmos-sdk/x/auth/vesting"
+  "github.com/cosmos/cosmos-sdk/x/bank"
+  bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+  banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+  "github.com/cosmos/cosmos-sdk/x/capability"
+  capabilitykeeper "github.com/cosmos/cosmos-sdk/x/capability/keeper"
+  capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+...
+```
+
+The modules in the `/cosmos-sdk/x/` folder are maintained by the Cosmos team. To understand a module, the best way is to have a look at the according `spec` folder. Have a look at the [`cosmos-sdk/x/bank/spec/01_state.md`](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/x/bank/spec/01_state.md) to understand the state of the bank module which you used in this section.
```

Comment:
Clarifying the decentralised workflow.
```suggestion
The modules in the `/cosmos-sdk/x/` folder are maintained by several organisations working on the Cosmos stack. To understand a module, the best way is to have a look at the respective `spec` folder. For example, have a look at the [`cosmos-sdk/x/bank/spec/01_state.md`](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/x/bank/spec/01_state.md) to understand the state of the bank module which you used in this section.
```

Also, would really appreciate a quick wrap-up of what was learned and a call to action under "Next Up". Not critical for beta-launch!